### PR TITLE
Added doctrine Proxy matcher/filter

### PR DESCRIPTION
**Pull request by malc0mn**
#18 when reading the Doctrine proxy issue, I felt quite stupid/embarrassed especially as @mnapoli states in [his comment](https://github.com/myclabs/DeepCopy/issues/18#issuecomment-151157194) to solve this by adding proxy filters/matchers...

So instead of nagging along, I tried it and it solved the issues I personally had with it. So I'm submitting this as a pull request **even though the classes feel too lightweight, as if I'm forgetting something**.
Please review and test properly first ;-)

### CODE REVIEW

**Comment by malc0mn** [comment]
Oh, and the way I use them is:

```php
<?php

$deepCopy = new DeepCopy();

// Load proxy classes. Add this as the very first filter!
$deepCopy->addFilter(new DoctrineProxyFilter(), new DoctrineProxyMatcher());
```

**Comment by Koc** [comment]
@malc0mn looks realy good. Can you update docs also, please?

File: src/DeepCopy/Matcher/Doctrine/DoctrineProxyMatcher.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Match a specific property of a specific class
+ */
+class DoctrineProxyMatcher implements Matcher
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($object, $property)
+    {
+        return ($object instanceof \Doctrine\Common\Persistence\Proxy);
```
Comment (Koc) [inline src/DeepCopy/Matcher/Doctrine/DoctrineProxyMatcher.php]: brackets not needed here and you can import class

Comment (malc0mn) [inline src/DeepCopy/Matcher/Doctrine/DoctrineProxyMatcher.php]: Will attend to that and the docs!

I have no idea how to write a test for this though :-s

**Comment by juulcode** [comment]
@malc0mn your code did exactly what I was looking for +1. It would be nice if this pull request could be merged. Are the docs up-to-date?

**Comment by mnapoli** [comment]
Thanks for the contribution! Writing tests isn't easy because Doctrine proxy are internal objects, I'll have a look at it. In any case those are 2 lines of code (really), if it doesn't work we'll hear about it, there is more value in shipping a solution for this year old problem. And it will not break anything since it's 2 new classes.

**Comment by knifesk** [comment]
Very cool @malc0mn, we all appreciate your time and effort to fix this issue! :)

**Comment by malc0mn** [comment]
You're welcome @knifesk, @mnapoli and @juulcode! Glad to be of service ;)

**Pull request merged**